### PR TITLE
[Patch] Auto scroll for order stepper

### PR DESCRIPTION
**Pull request by ds300**
Following on from #1574 (see the comment at the end)

## Here's the fix

image

I needed to refactor the `Tabs` and `RouteTabs` to let the component be scrollable on mobile without sacrificing the border-bottom on the active tab. This is actually a bug in prod right now:

## No border-bottom on active tabs in prod 😧 

no-border-bottom

## this PR fixes that 🎈🎉 

(using an extra couple of layers of div, of course)

border-bottom-yay

Also cleaned up the `RouteTabs` component while I was digging around in there.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 410f510dff59f058729720c0839ebccda54218ca

https://deploy-preview-1583--artsy-reaction.netlify.com

File: src/Apps/Artist/Components/NavigationTabs.tsx [hunk 1]
```diff
@@ -91,16 +88,9 @@ export class NavigationTabs extends React.Component<Props> {
...
-          <RouteTabs mx={-4} size={"xs"}>
-            {this.renderTabs("xs")}
-          </RouteTabs>
-        </Media>
-        <Media greaterThan="xs">
-          <RouteTabs mx={0} size={null}>
-            {this.renderTabs()}
-          </RouteTabs>
-        </Media>
+        <Flex mx={[-2, 0]}>
```
Comment (damassi) [inline src/Apps/Artist/Components/NavigationTabs.tsx]: Much nicer 😄

File: src/Styleguide/Components/Tabs.tsx [hunk 2]
```diff
@@ -75,14 +82,64 @@ export class Tabs extends React.Component<TabsProps, TabsState> {
...
+
+    const activeTabOffset = activeTabRect.left - containerRect.left
+
+    const desiredActiveTabOffset =
+      containerRect.width / 2 - activeTabRect.width / 2
+
+    const offsetDiff = activeTabOffset - desiredActiveTabOffset
+
+    const currentScroll = this.containerRef.scrollLeft
+
```
Comment (damassi) [inline src/Styleguide/Components/Tabs.tsx]: #minor maybe remove the empty lines between the vars?

File: src/Styleguide/Components/__stories__/Tabs.story.tsx [hunk 3]
```diff
@@ -97,6 +97,27 @@ storiesOf("Styleguide/Components", module).add("Tabs (Simple)", () => {
...
+          <Tab name="Funicular" />
+          <Tab name="DiCaprio" />
+          <Tab name="Fathomable" />
+          <Tab name="Shiny" />
+          <Tab name="Cyprus" />
+          <Tab name="UMass" />
+          <Tab name="Skiing" />
+          <Tab name="Liver" />
+          <Tab name="Magazine" />
+        </Tabs>
```
Comment (damassi) [inline src/Styleguide/Components/__stories__/Tabs.story.tsx]: All the nice thingsss :)

**Comment by damassi** [comment]
merge on green

**Comment by ds300** [comment]
merge on green ? Is peril alive?

**Pull request merged**